src/Symfony/Component/Finder/Iterator/FilterIterator.php

```diff
@@ -18,9 +18,24 @@
  *
  * @author Alex Bogomazov
  */
-abstract class FilterIterator extends \FilterIterator
+abstract class FilterIterator extends \FilterIterator implements \RecursiveIterator
```

Comment:
I'm not sure implementing RecursiveIterator on all out filters is a good idea. Most of our filters are required to be applied on a flattened iterator to work properly.

What about using a child class RecursiveFilterIterator which would become the parent of the filters applied in a recursive way ?